### ISSUE TITLE
BioPerl subroutine warning message and intergenic_variant annotation

### ISSUE THREAD

**Issue by arthurmelobio** [Ensembl/ensembl-vep#69]
Dear VEP Team, 

I have tried to functionally annotate SNPs between two close related wheat genomes. So, I'm using a own genome sequence fasta file as well the GFF file. I had processed the GFF file and then run vep with the follow command line.

`grep -v "#" transcripts.CDS.gff | sort -k1,1 -k4,4n -k5,5n | bgzip -c > transcripts.CDS.gff.gz`

`tabix -p off transcripts.CDS.gff.gz`

`/path/to/vep/vep -i input.vcf --format vcf -o VEPout.txt --force_overwrite --variant_class --regulatory --symbol --stats_text --fasta transcripts.CDS.fasta --gff transcripts.CDS.gff.gz`

I'm getting a lot o warning messages related with a perl subroutine and BioPerl:

`Deep recursion on subroutine "Bio::EnsEMBL::VEP::Parser::VCF::create_VariationFeatures" at /home/arthur/Bioinfo_Softwares/ensembl-vep/modules/Bio/EnsEMBL/VEP/Parser/VCF.pm line 261, <__ANONIO__> line 4247032.`

Even with this warning, the output is created but all variants are annotated as "intergenic_variant". However, running vep through --cache and sellecting a database using --species, the analysis doen't print the message above and the output looks better with ~ 94% of variants classified into "intergenic_variant".

So, my questions are:

1. What this Perl subroutine warning message means? Why it appear only on analysis using FASTA + GFF? How it can affect the results?

2. The "intergenic_variants" are assigned for variants which failed in annotation? In analysis where all the variants are annotated as "intergenic_variants" I got some problem in a database creation or the GFF file has no enough information to more detail annotation?

3. How can I create my own database on ~/.vep?

Thank you very much for the help.

Best,
Arthur

**Comment by willmclaren** [comment]
My guess is that the FASTA you have specified contains transcript-level sequence, not genomic sequence. VEP is not currently capable of using this setup; the FASTA file must be the genomic sequence. See [docs](http://www.ensembl.org/info/docs/tools/vep/script/vep_cache.html#gff).

Further to this, it may be that the GFF and VCF (and/or FASTA) have different chromosome names; the VEP cache contains a synonyms file (`$HOME/.vep/[species]/[version]_[assembly]/chr_synonyms.txt`) which you may manually use with [`--synonyms`](http://www.ensembl.org/info/docs/tools/vep/script/vep_options.html#opt_synonyms). This will allow VEP to map different names for chromosomes

**Comment by arthurmelobio** [comment]
Hi @willmclaren, 

Thank you for your quick reply. 

Unfortunately I have the same error by using the genome-based files: reference (FASTA) and annotation (GTF). My chromosomes IDs are the same among all three files (VCF, GTF and FASTA). I'm guessing the issue is on input VCF file once the VCF.pm script is used to parse the VCF file. Right?

I took a look on **ensembl-vep/modules/Bio/EnsEMBL/VEP/Parser/VCF.pm** script and I guess the 
issue is to parse the alternative allele from the VCF file. Trying to get more detailed information, I used `use diagnostics;` and I got

`Use of uninitialized value $ref in join or string at /home/arthur/Bioinfo_Softwares/ensembl-vep/modules/Bio/EnsEMBL/VEP/Parser/VCF.pm line 236 (#1)`

`(W uninitialized) An undefined value was used as if it were already defined.  It was interpreted as a "" or a 0, but maybe it was a mistake. To suppress this warning assign a defined value to your variables.`

`To help you figure out what was undefined, perl will try to tell you the name of the variable (if any) that was undefined.  In some cases it cannot do this, so it also tells you what operation you used the undefined value in.  Note, however, that perl optimizes your program and the operation displayed in the warning may not necessarily appear literally in your program.  For example, "that $foo" is usually optimized into "that " . $foo, and the warning will refer to the concatenation (.) operator, even though there is no . in your program.`

`Use of uninitialized value in addition (+) at /home/arthur/Bioinfo_Softwares/ensembl-vep/Bio/EnsEMBL/IO/Parser/BaseVCF4.pm line 267 (#1)`
`Use of uninitialized value in split at /home/arthur/Bioinfo_Softwares/ensembl-vep/Bio/EnsEMBL/IO/Parser/BaseVCF4.pm line 416 (#1)`
`Use of uninitialized value in string eq at /home/arthur/Bioinfo_Softwares/ensembl vep/modules/Bio/EnsEMBL/VEP/Parser/VCF.pm line 257 (#1)`
`Use of uninitialized value $ref in concatenation (.) or string at /home/arthur/Bioinfo_Softwares/ensembl-vep/modules/Bio/EnsEMBL/VEP/Parser/VCF.pm line 337 (#1)`
`Use of uninitialized value in sprint at /home/arthur/Bioinfo_Softwares/ensembl-vep/modules/Bio/EnsEMBL/VEP/Parser.pm line 505 (#1)`
`Use of uninitialized value in concatenation (.) or string at /home/arthur/Bioinfo_Softwares/ensembl-vep/modules/Bio/EnsEMBL/VEP/Parser.pm line 514 (#1)`

My VCF is v.4.2 and it looks like:

chr1A	40613	.	T	.	33.5884	.	DP=1;MQ0F=1;AN=2;DP4=0,1,0,0;MQ=0	GT	0/0
chr1A	40614	.	T	.	33.5884	.	DP=1;MQ0F=1;AN=2;DP4=0,1,0,0;MQ=0	GT	0/0
chr1A	40615	.	A	.	33.5884	.	DP=1;MQ0F=1;AN=2;DP4=0,1,0,0;MQ=0	GT	0/0
chr1A	40616	.	A	.	33.5884	.	DP=1;MQ0F=1;AN=2;DP4=0,1,0,0;MQ=0	GT	0/0
chr1A	40617	.	T	.	33.5884	.	DP=1;MQ0F=1;AN=2;DP4=0,1,0,0;MQ=0	GT	0/0

Sorry for this long post, but I'm trying to provide all informations.
Any advice?

Thank you.

**Comment by arthurmelobio** [comment]
Well... I guess I figured out the issue. 

The VCF file has some troubles related with version and INFO field. The version needs to be 4.0 and after noticed the INFO field reports different info's for different variants, I decided keep this field blank as all vcf input from the example folder. I also remove all monomorphic variants from my vcd input.
So, the analysis runned without the subroutine warning message, however I found a new error. I'm gonna open other thread to discuss it.

Thank you
